### PR TITLE
Fix for VENOM vulnerability (CVE-2015-3456)

### PR DESCRIPTION
**Pull request by danmcd**
Three variants-on-a-theme patches were issued via the oss-security list this morning.  This is a direct application of xsa133-qemuu.patch (using "patch < xsa133-qemuu.patch" with PWD=illumos-kvm-cmd/hw).

### CODE REVIEW

**Comment by rmustacc** [comment]
Resolved in 407546e. Thanks Dan!

**Comment by misterbisson** [comment]
Thanks for the quick patch @danmcd.

It's been noted elsewhere, but worth repeating here:

[From Alex](http://www.listbox.com/member/archive/184463/2015/05/sort/time_rev/page/1/entry/1:60/20150513102223:76C2E8B4-F97B-11E4-97ED-F81F8C36314E/):

> One other thing to note is that the qemu process on SmartOS runs inside a zone, so escaping the qemu just gets you root in a zone that has basically nothing in it except the qemu binary and some config.

> You would need an additional kernel privesc vuln to escape that zone and take control over the entire box.

And [from Robert](http://www.listbox.com/member/archive/184463/2015/05/sort/time_rev/page/1/entry/0:60/20150513113241:4894DE98-F985-11E4-8012-CC6735A537A4/)

> This is correct, the processes in there are running in a stripped privilege environment. It cannot fork.

tl;dr: this vulnerability poses no risk on SmartOS.

**Comment by misterbisson** [comment]
Support statement regarding Joyent's public and private cloud offerings: https://help.joyent.com/entries/68099220-Security-Advisory-on-Venom-CVE-2015-3456-in-KVM-QEMU
